src/hooks/core/chat/useDMMessage.ts

```diff
@@ -0,0 +1,112 @@
+import {v4 as uuid} from 'uuid';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import ChannelListMemberSchema from '../../../database/schema/ChannelListMemberSchema';
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import ChatSchema from '../../../database/schema/ChatSchema';
+import UserSchema from '../../../database/schema/UserSchema';
+import {DEFAULT_PROFILE_PIC_PATH} from '../../../utils/constants';
+import useUserAuthHook from '../auth/useUserAuthHook';
+import useChatUtilsHook from './useChatUtilsHook';
+import {initChatFromPost, initChatFromPostAnon} from '../../../service/chat';
+
+type ChannelCategory = 'SIGNED' | 'ANONYMOUS';
+
+const useDMMessage = () => {
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const {signedProfileId, anonProfileId} = useUserAuthHook();
+  const {goToChatScreen} = useChatUtilsHook();
+
+  const saveChannelList = async (channel, type: ChannelCategory) => {
+    try {
+      if (channel?.members?.length === 0) return Promise.reject(Error('no members'));
+      let userTarget = channel?.better_channel_members.find(
+        (data) => data?.user?.id !== signedProfileId
+      );
+
+      if (type === 'ANONYMOUS') {
+        userTarget = channel?.better_channel_members.find(
+          (data) => data?.user?.id !== anonProfileId
+        );
+      }
+
+      const chatType = type === 'ANONYMOUS' ? 'ANON_PM' : 'PM';
+      const chatName = {
+        name: userTarget?.user?.name || userTarget?.user?.username,
+        image: userTarget?.user?.image || DEFAULT_PROFILE_PIC_PATH
+      };
+
+      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      channel.myUserId = signedProfileId;
+      channel.targetName = chatName?.name;
+      channel.targetImage = chatName?.image;
+
+      channel.channel = {...channel};
+      const channelList = ChannelList.fromChannelAPI(channel, chatType, channel?.members);
+      await channelList.saveIfLatest(localDb);
+      refresh('channelList', 'chat', 'channelInfo', 'channelMember');
+      return goToChatScreen(channelList);
+    } catch (error) {
+      console.log('sendMessage', error);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `saveChannelList` is doing more than just saving the channel list. It's also modifying the channel object and navigating to the chat screen. This violates the Single Responsibility Principle (SRP) of SOLID principles. Consider breaking this function into smaller ones, each with a single responsibility.

```diff
+ const modifyChannel = (channel, type) => {
+   let userTarget = channel?.better_channel_members.find(
+     (data) => data?.user?.id !== signedProfileId
+   );
+
+   if (type === 'ANONYMOUS') {
+     userTarget = channel?.better_channel_members.find(
+       (data) => data?.user?.id !== anonProfileId
+     );
+   }
+
+   const chatType = type === 'ANONYMOUS' ? 'ANON_PM' : 'PM';
+   const chatName = {
+     name: userTarget?.user?.name || userTarget?.user?.username,
+     image: userTarget?.user?.image || DEFAULT_PROFILE_PIC_PATH
+   };
+
+   channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+   channel.myUserId = signedProfileId;
+   channel.targetName = chatName?.name;
+   channel.targetImage = chatName?.image;
+
+   channel.channel = {...channel};
+   return channel;
+ }

- const saveChannelList = async (channel, type: ChannelCategory) => {
+ const saveChannelList = async (channel) => {
    try {
      if (channel?.members?.length === 0) return Promise.reject(Error('no members'));
-     let userTarget = channel?.better_channel_members.find(
-       (data) => data?.user?.id !== signedProfileId
-     );
-
-     if (type === 'ANONYMOUS') {
-       userTarget = channel?.better_channel_members.find(
-         (data) => data?.user?.id !== anonProfileId
-       );
-     }
-
-     const chatType = type === 'ANONYMOUS' ? 'ANON_PM' : 'PM';
-     const chatName = {
-       name: userTarget?.user?.name || userTarget?.user?.username,
-       image: userTarget?.user?.image || DEFAULT_PROFILE_PIC_PATH
-     };
-
-     channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
-     channel.myUserId = signedProfileId;
-     channel.targetName = chatName?.name;
-     channel.targetImage = chatName?.image;
-
-     channel.channel = {...channel};
      const channelList = ChannelList.fromChannelAPI(channel, chatType, channel?.members);
      await channelList.saveIfLatest(localDb);
      refresh('channelList', 'chat', 'channelInfo', 'channelMember');
-     return goToChatScreen(channelList);
    } catch (error) {
      console.log('sendMessage', error);
    }
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->